osu.Game/Updater/AssociationManager.cs

```diff
@@ -0,0 +1,168 @@
+        private void load()
+        {
+            Logger.Log("Checking file associations");
+
+            bool fileAssociationStatus = EnsureAssociationsSet();
+
+            if (!fileAssociationStatus)
+            {
+                // Files are not associated, prompt the user to fix them
+                Notifications.Post(new SimpleNotification
+                {
+                    Icon = FontAwesome.Solid.ExclamationCircle,
+                    Text = "File associations are not set up correctly! Do you want to fix them?",
+                    Activated = () =>
+                    {
+                        InitializeFileAssociations();
+                        Notifications.Post(new SimpleNotification
+                        {
+                            Icon = FontAwesome.Solid.CheckCircle,
+                            Text = "File associations have been fixed!",
+                        });
+                        return true;
+                    }
+                });
+            }
+        }
+
+        /// <summary>
+        /// Initializes (creates) the file associations
+        /// </summary>
+        public void InitializeFileAssociations()
+        {
+            Logger.Log("Setting up file associations!");
+            string programPath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location) + "\\osu!.exe";
+
+            foreach (string extension in associated_extensions)
+            {
+                CreateFileAssociation(extension, programPath);
+            }
+
+            SHChangeNotify(0x8000000, 0x1000, IntPtr.Zero, IntPtr.Zero); // Notify Explorer that the file associations have changed
+        }
+
+        /// <summary>
+        /// Create an association for a specific extension
+        /// </summary>
+        /// <param name="extension"></param>
+        /// <param name="programPath"></param>
+        /// <exception cref="InvalidOperationException"></exception>
+        public void CreateFileAssociation(string extension, string programPath)
+        {
+            Logger.Log("Creating file association for " + extension);
+
+            // Check if the extension has a file explorer override, if so, delete the override
+            RegistryKey? fileExplorerKey = Registry.CurrentUser.OpenSubKey("Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\FileExts\\" + extension, true);
+
+            if (fileExplorerKey != null)
+            {
+                fileExplorerKey.DeleteSubKey("UserChoice", false);
+                fileExplorerKey.Close();
+                Logger.Log("Deleted file explorer override for " + extension);
+            }
+            else
+            {
+                Logger.Log("File explorer override for " + extension + " does not exist!");
+            }
+
+            RegistryKey key = Registry.CurrentUser.CreateSubKey($"Software\\Classes\\{extension}");
+            key.CreateSubKey("shell\\\\open\\\\command").SetValue("", $"\"{programPath}\" \"%1\"");
```

Comment:
Are the quadruple backslashes actually necessary? All the other `*SubKey()` calls only use double backslashes.